src/main/scala/io/iohk/ethereum/consensus/validators/std/StdOmmersValidator.scala

```diff
@@ -144,19 +152,17 @@ class StdOmmersValidator(blockchainConfig: BlockchainConfig, blockHeaderValidato
       parentHash: ByteString,
       blockNumber: BigInt,
       getNBlocksBack: GetNBlocksBack
-  ): Option[Seq[BlockHeader]] = {
+  ): Seq[BlockHeader] = {
     val numberOfBlocks = blockNumber.min(OmmerGenerationLimit).toInt
-    val ancestors = getNBlocksBack(parentHash, numberOfBlocks).map(_.header)
-    Some(ancestors).filter(_.length == numberOfBlocks)
+    getNBlocksBack(parentHash, numberOfBlocks).map(_.header)
   }
 
   private def collectOmmersFromAncestors(
       parentHash: ByteString,
       blockNumber: BigInt,
       getNBlocksBack: GetNBlocksBack
-  ): Option[Seq[BlockHeader]] = {
+  ): Seq[BlockHeader] = {
     val numberOfBlocks = blockNumber.min(OmmerGenerationLimit).toInt
-    val ancestors = getNBlocksBack(parentHash, numberOfBlocks).map(_.body.uncleNodesList)
-    Some(ancestors).filter(_.length == numberOfBlocks).map(_.flatten)
+    getNBlocksBack(parentHash, numberOfBlocks).flatMap(_.body.uncleNodesList)
```

Comment:
I have simplified both `collectAncestors` and `collectOmmersFromAncestors` methods, by removing the final check.
I was checking that the `getNBlocksBack` did return the expected number of items, otherwise it would just return None.
Is there any reason to keep such a check?

Reply:
Ideally this would've been covered by tests if it was important. I can't comment on the check itself.

What leads you to believe it's not necessary to keep? Perhaps we can leave an assert in place, just in case?

Reply:
I cannot find a single reason to keep it, it adds noise to the code. The method was returning an `Option[List]` which is quite redundant and makes everything more hard to read

Reply:
I wouldn't remove it...in the worst case if `getNBlocksBack` returns an empty list, the ommers validation will return `Right`.

The `validateOmmersAncestors` method used to return a `OmmersAncestorsError` to indicate there was a problem getting the ancestors...that behavior seems to be lost as well.

Reply:
`validateOmmersAncestors` was returning `OmmersUsedBeforeError`, but since two different validations were happening in that method I made it clear by replacing `OmmersUsedBeforeError` by two others that are more precise.

Reply:
`validateOmmersAncestors` was returning `OmmersAncestorsError` in case `getNBlocksBack` returned an invalid number of results: https://github.com/input-output-hk/mantis/pull/948/files#diff-4ffb3286be2984d1679ee76bbb2e5e404f64147f1d77ab0eb95f8031a1379635L107

Reply:
Ah, ok, now I understand your point. The `getNBlocksBack` is a method that is written by Mantis, so I don't understand why we would have to validate it every single time it's called.

Reply:
Ideally `getNBlocksBack` would return a type that allows encoding errors but since it doesn't, it's a good idea to check whether it returns the expected number of blocks.

In theory we should always be able to get the expected numbers of blocks, so if the number doesn't match something is definitely broken. By removing the check and defaulting to `Right` that problem will be silently ignored.

Reply:
So I am adding a check in `private def validateOmmersNotUsed`,making sure that is the list is empty the validation fails, in `private[std] def validateOmmersAncestors` the validation will already fail if the list of ancestors is empty

Reply:
I have also added a test names `report failure if getNBlocksBack returns an empty list of ancestors`

Reply:
I had to remove the explicit check that checked that the list of ommer's ancestors as not empty, because it was causing 2 ETS tests to fails

Reply:
We shouldn't ignore the issue if blocks aren't found. I'd suggest one of the two approaches

- add a check for the correct size of the result and return an appropriate error otherwise
- merge as it is but create a follow-up ticket to refactor `GetNBlocksBack` to return an error in case blocks aren't found (better approach imho)

Reply:
I created a follow-up ticket ETCM-776